newrelic/data_source_newrelic_synthetics_monitor.go

```diff
@@ -0,0 +1,83 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+
+	synthetics "github.com/dollarshaveclub/new-relic-synthetics-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceNewRelicSyntheticsMonitor() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNewRelicSyntheticsMonitorRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_API_KEY", nil),
+				Sensitive:   true,
+			},
+			"max_check": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+			},
+		},
+	}
+}
+
+func dataSourceNewRelicSyntheticsMonitorRead(d *schema.ResourceData, meta interface{}) error {
+
+	apiKey := d.Get("api_key").(string)
```

Comment:
Typically you would set this up in the provider.go file. You can have more than one client on the meta, that is commonly do in other providers, especially since this `api_key` is already defined at the provider level.

Reply:
This is what I mean by putting multiple clients on the meta object, unfortunately its a bit of a refactor: https://github.com/terraform-providers/terraform-provider-azurerm/blob/91e4562440f9a8a1bb5ea58947e28dd1134d9e63/azurerm/config.go#L73-L76

```go
type providerClient struct {
  Client *newrelic.Client
  Synthetics *synthetics.Client
}
```

You basically just create a new client struct that would have both this existing client plus the new synthetics client to use as the meta in your resource methods. Unfortunately it means you have to adjust the [type assertions](https://github.com/terraform-providers/terraform-provider-newrelic/blob/744ea7cf38878334670a690094ac6e0a278d771c/newrelic/resource_newrelic_alert_channel.go#L107) to be the new struct you create.

But this way you can use the provider config to setup both clients at the same time.

Reply:
The recent changes merged in v`1.0.1` introduced a `ProviderClient` as above.